instrumentation/kafka-streams/src/main/java/brave/kafka/streams/KafkaStreamsTracing.java

```diff
@@ -121,6 +129,132 @@ public KafkaStreams kafkaStreams(Topology topology, Properties streamsConfig) {
     return new TracingValueTransformerWithKeySupplier<>(this, name, valueTransformerWithKey);
   }
 
+  /**
+   * Create a foreach processor, similar to {@link KStream#foreach(ForeachAction)}, where its action
+   * will be recorded in a new span with the indicated name.
+   *
+   * <p>Simple example using Kafka Streams DSL:
+   * <pre>{@code
+   * StreamsBuilder builder = new StreamsBuilder();
+   * builder.stream(inputTopic)
+   *        .process(kafkaStraemsTracing.foreach("myForeach", (k, v) -> ...);
+   * }</pre>
+   */
+  public <K, V> ProcessorSupplier<K, V> foreach(String name, ForeachAction<K, V> action) {
```

Comment:
Maybe this has been discussed already but I don't find the parameter name to be explicit enough. In IDE, we are not sure if it is the span name or something else.
Could we rename it to spanName for example ?

Reply:
Totally agree 👍 `spanName` LGTM.